src/main/java/org/cyclonedx/generators/json/BomJsonGenerator12.java

```diff
@@ -55,16 +68,49 @@ public Version getSchemaVersion() {
      * @since 3.0.0
      */
     public JSONObject generate() {
-        doc.put("bomFormat", "CycloneDX");
-        doc.put("specVersion", Version.VERSION_12.getVersionString());
-        if (bom.getSerialNumber() != null) {
-            doc.put("serialNumber", bom.getSerialNumber());
+        try {
+            doc = new JSONObject(toJson(this.bom, false));
+
+            return doc;
+        }
+        catch (GeneratorException e) {
+            return null;
```

Comment:
I think it's better to throw an exception here, instead of null

Reply:
What I was trying to do is maintain the original API's method signature. I'd rather throw it too, but I'm kinda like ehhhhhh.